### PR TITLE
[v0.6] Refactored texture kinds

### PR DESCRIPTION
**Pull request by kvark**
Partially contains #756

### CODE REVIEW

File: src/device/tex.rs [hunk 1]
```diff
@@ -300,55 +339,55 @@ pub enum FilterMethod {
...
 /// restricted by what Metal exposes, though this could conceivably be
 /// extended in the future. Note that a single texture can *only* ever be of
 /// one kind. A texture created as `Texture2D` will forever be `Texture2D`.
 // TODO: "Texture views" let you get around that limitation.
 #[derive(Eq, Ord, PartialEq, PartialOrd, Hash, Copy, Clone, Debug)]
-pub enum TextureKind {
+pub enum Kind {
     /// A single row of texels.
-    Texture1D,
+    D1,
```
Comment (csherratt) [inline src/device/tex.rs]: YAY \o/. This is much less verbose.

Comment (kvark) [inline src/device/tex.rs]: I was just summing you in gitter to ask what you think about it. `Dx` pattern may be a little confusing at first, but definitely very short to type.

Comment (csherratt) [inline src/device/tex.rs]: I'm ok with it. I can't think of a better name that is short, and legal.

Comment (kvark) [inline src/device/tex.rs]: \o/

**Pull request merged**